stdlib/3/multiprocessing/connection.pyi

```diff
@@ -7,6 +7,7 @@ import types
 _Address = Union[str, Tuple[str, int]]
 
 class _ConnectionBase:
+    def __init__(self, handle: int, readable: bool = ..., writable: bool = ...) -> None: ...
```

Comment:
The code goes out of its way to make this support any object with `__index__`, so the annotation for handle should be `typing.SupportsIndex`.

Reply:
This is a good distinction. After looking at the constructor of `_ConnectionBase` I realized the first statement was `handle = handle.__index__()`. Thanks !

Reply:
Now I see that `SupportsIndex` is supported only on version 3.8. I suggest to revert this change and allow it to stay `int`.

Reply:
Normally I'd say, you can import `SupportsIndex` from `typing_extensions`, but it isn't included (python/typing#723). But we should at least use `SupportsIndex` on Python >= 3.8 using a version check.

Reply:
Added version check on the latest commit. The build has failed due to 429 error when trying to fetch the repository :/